python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -582,11 +739,137 @@ def rank(x):
         return result
 
     @cached_property
-    def _groupby(self):
-        return libgroupby.GroupBy(
-            [*self.grouping.keys._columns], dropna=self._dropna
+    def _groupby(self) -> types.SimpleNamespace:
+        with acquire_spill_lock() as spill_lock:
+            plc_groupby = plc.groupby.GroupBy(
+                plc.Table(
+                    [
+                        col.to_pylibcudf(mode="read")
+                        for col in self.grouping.keys._columns
+                    ]
+                ),
+                plc.types.NullPolicy.EXCLUDE
+                if self._dropna
+                else plc.types.NullPolicy.INCLUDE,
+            )
+            # Do we need this because we just check _spill_locks in test_spillable_df_groupby?
```

Comment:
Should we rephrase this as a note?
```suggestion
            # Note: We return a SimpleNamespace with an additional _spill_locks
            # attribute solely to verify that the spill_lock is set correctly.
```

Reply:
Going to keep this one phrased as a question. I _think_ if this is only used for testing (and is maybe unnecessary) we could potentially just remove this in the future